geomstats/geometry/manifold.py

```diff
@@ -121,6 +151,43 @@ def equip_with_quotient_structure(self):
         self.quotient = self.new(equip=False)
         self.quotient.equip_with_metric(QuotientMetric_, fiber_bundle=self.fiber_bundle)
 
+    def equip_with_structure(self, Structure, **kwargs):
+        """Endow manifold with mathematic structure.
+
+        Methods available in `Structure` become directly available to
+        the manifold.
+
+        A space can be endowed with several math structures. In case of
+        name clashing, first endowed structure has priority.
+
+        User is responsible for compatibility between mathematical
+        structures.
+
+        Parameters
+        ----------
+        Structure : Structure object
+        """
+        self._attr_cache = {}
+        self._structures.append(Structure(self, **kwargs))
+        return self
+
+    def equip_maximally(
```

Comment:
what is the difference between this method and the one above? Could you add an explanatory docstring?